### PR TITLE
Add OTP 21 support and CI coverage

### PR DESCRIPTION
**Pull request by getong**

### CODE REVIEW

**Comment by uwiger** [comment]
Sorry for the delay. I have no objections, other than that perhaps we should test 21.0.5 instead?

**Comment by tolbrino** [comment]
@uwiger Looks like it would be better to get #101 merged first?

**Comment by getong** [comment]
@uwiger 
test 21.0.5 is updated.

**Comment by tolbrino** [comment]
@getong Could you rebase on the latest master? Also please squash your commits into one commit.

**Comment by tolbrino** [comment]
> Yes, I will, but tomorrow.

That's fine. I'll wait on that.

**Comment by getong** [comment]
@tolbrino 
The code is done.

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-REBAR3=$(shell which rebar3 || echo ./rebar3)
```
Comment (tolbrino) [inline Makefile]: I think it would make sense to remove the `rebar3` binary from the repo altogether. 

@uwiger Thoughts?

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+REBAR3=./rebar3
 
 .PHONY: all clean clean_plt compile test doc dialyzer xref ci
 
 EXOMETER_PLT=exometer_core.plt
 DIALYZER_OPTS = # -Wunderspecs
 DIALYZER_APPS = erts kernel stdlib compiler syntax_tools \
-		test_server common_test folsom \
```
Comment (tolbrino) [inline Makefile]: Why did you remove `test_server`?

Comment (tolbrino) [reply Makefile]: That has only happened in 19.0, however we still support 18.3 so we should keep that.

File: Makefile (hunk 1 above)
Comment (tolbrino) [reply Makefile]: For Travis CI it makes more sense to get the latest stable version of `rebar3` as part of the CI step. This way we wouldn't have to maintain a version of `rebar3` in the repository.

**Comment by getong** [comment]
@tolbrino 
The rebar3 file is deleted, and the Makefile use the local rebar3 if possbile.

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-REBAR3=$(shell which rebar3 || echo ./rebar3)
+REBAR3=$(shell [ -e ./rebar3 ] && echo ./rebar3 || which rebar3)
```
[the remark this answers is not in the document]
Comment (getong) [reply Makefile]: done

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,11 @@
 sudo: false
 language: erlang
-script: "make ci"
+script:
+    - wget -c https://github.com/erlang/rebar3/releases/download/3.6.2/rebar3
+    - chmod +x rebar3
+    - make ci
```
[the remark this answers is not in the document]
Comment (getong) [reply .travis.yml]: done

**Comment by tolbrino** [comment]
@getong Thanks for finishing the changes.
